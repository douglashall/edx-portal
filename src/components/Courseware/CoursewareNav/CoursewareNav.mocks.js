const mockCourseOutline = {
  displayName: 'Demo Course',
  type: 'course',
  id: 'DemoCourse',
  descendants: [
    {
      displayName: 'Chapter 1',
      type: 'chapter',
      id: 'c1',
      descendants: [
        {
          displayName: 'Section 1.1',
          type: 'sectional',
          id: 's1.1',
          descendants: [
            {
              displayName: 'Vertical 1.1.1',
              type: 'vertical',
              id: 'v1.1.1',
              descendants: [],
            },
            {
              displayName: 'Vertical 1.1.2',
              type: 'vertical',
              id: 'v1.1.2',
              descendants: [],
            },
            {
              displayName: 'Vertical 1.1.3',
              type: 'vertical',
              id: 'v1.1.3',
              descendants: [],
            },
            {
              displayName: 'Vertical 1.1.4',
              type: 'vertical',
              id: 'v1.1.4',
              descendants: [],
            },
          ],
        },
        {
          displayName: 'Section 1.2',
          type: 'sectional',
          id: 's1.2',
          descendants: [
            {
              displayName: 'Vertical 1.2.1',
              type: 'vertical',
              id: 'v1.2.1',
              descendants: [],
            },
          ],
        },
      ],
    },
    {
      displayName: 'Chapter 2',
      type: 'chapter',
      id: 'c2',
      descendants: [
        {
          displayName: 'Section 2.1',
          type: 'sectional',
          id: 's2.1',
          descendants: [
            {
              displayName: 'Vertical 2.1.1',
              type: 'vertical',
              id: 'v2.1.1',
              descendants: [],
            },
          ],
        },
      ],
    },
    {
      displayName: 'Chapter 3',
      type: 'chapter',
      id: 'c3',
      descendants: [
        {
          displayName: 'Section 3.1',
          type: 'sectional',
          id: 's3.1',
          descendants: [
            {
              displayName: 'Vertical 3.1.1',
              type: 'vertical',
              id: 'v3.1.1',
              descendants: [],
            },
          ],
        },
        {
          displayName: 'Section 3.2',
          type: 'sectional',
          id: 's3.2',
          descendants: [
            {
              displayName: 'Vertical 3.2.1',
              type: 'vertical',
              id: 'v3.2.1',
              descendants: [],
            },
          ],
        },
        {
          displayName: 'Section 3.3',
          type: 'sectional',
          id: 's3.3',
          descendants: [
            {
              displayName: 'Vertical 3.3.1',
              type: 'vertical',
              id: 'v3.3.1',
              descendants: [],
            },
            {
              displayName: 'Vertical 3.3.2',
              type: 'vertical',
              id: 'v3.3.2',
              descendants: [],
            },
          ],
        },
      ],
    },
  ],
};

/* eslint-disable import/prefer-default-export, object-curly-newline */
export {
  mockCourseOutline,
};
