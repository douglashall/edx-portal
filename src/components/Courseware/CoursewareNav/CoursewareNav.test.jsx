import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import CoursewareNav from './index';
import { mockCourseOutline } from './CoursewareNav.mocks';

describe('CoursewareNav', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create((
        <MemoryRouter>
          <CoursewareNav courseOutline={mockCourseOutline} />
        </MemoryRouter>))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
