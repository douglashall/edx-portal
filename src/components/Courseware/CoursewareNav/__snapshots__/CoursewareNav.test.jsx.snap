// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursewareNav renders correctly 1`] = `
<nav
  id="navigation"
>
  <h1>
    Demo Course
  </h1>
  <ul>
    <li>
      Chapter 1
      <ul>
        <li>
          Section 1.1
          <ul>
            <li>
              <a
                href="//v1.1.1"
                onClick={[Function]}
              >
                Vertical 1.1.1
              </a>
              <ul />
            </li>
            <li>
              <a
                href="//v1.1.2"
                onClick={[Function]}
              >
                Vertical 1.1.2
              </a>
              <ul />
            </li>
            <li>
              <a
                href="//v1.1.3"
                onClick={[Function]}
              >
                Vertical 1.1.3
              </a>
              <ul />
            </li>
            <li>
              <a
                href="//v1.1.4"
                onClick={[Function]}
              >
                Vertical 1.1.4
              </a>
              <ul />
            </li>
          </ul>
        </li>
        <li>
          Section 1.2
          <ul>
            <li>
              <a
                href="//v1.2.1"
                onClick={[Function]}
              >
                Vertical 1.2.1
              </a>
              <ul />
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      Chapter 2
      <ul>
        <li>
          Section 2.1
          <ul>
            <li>
              <a
                href="//v2.1.1"
                onClick={[Function]}
              >
                Vertical 2.1.1
              </a>
              <ul />
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      Chapter 3
      <ul>
        <li>
          Section 3.1
          <ul>
            <li>
              <a
                href="//v3.1.1"
                onClick={[Function]}
              >
                Vertical 3.1.1
              </a>
              <ul />
            </li>
          </ul>
        </li>
        <li>
          Section 3.2
          <ul>
            <li>
              <a
                href="//v3.2.1"
                onClick={[Function]}
              >
                Vertical 3.2.1
              </a>
              <ul />
            </li>
          </ul>
        </li>
        <li>
          Section 3.3
          <ul>
            <li>
              <a
                href="//v3.3.1"
                onClick={[Function]}
              >
                Vertical 3.3.1
              </a>
              <ul />
            </li>
            <li>
              <a
                href="//v3.3.2"
                onClick={[Function]}
              >
                Vertical 3.3.2
              </a>
              <ul />
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;
